blockstore.go

```diff
@@ -161,7 +161,11 @@ func (b *Blockstore) AllKeysChan(ctx context.Context) (<-chan cid.Cid, error) {
 				if mh, err := base64.RawStdEncoding.DecodeString(mh); err != nil {
 					log.Printf("failed to parse multihash when querying all keys in sqlite3 blockstore: %s", err)
 				} else {
-					ret <- cid.NewCidV1(codec, mh)
+					select {
+					case ret <- cid.NewCidV1(codec, mh):
+					case <-ctx.Done():
+						return
+					}
 				}
 			case ctx.Err() != nil:
```

Comment:
IMO this ctx check is now kinda redundant, as it's surrounded by another two "are we cancelled" checks.

Reply:
And in turn you could also replace the naked switch with just a good old `if err != nil` check.

Reply:
IMO this ctx check is now kinda redundant, as it's surrounded by another two "are we cancelled" checks.

It's to avoid logging in that case. But I agree.